extensions/smallrye-rest-client/deployment/src/main/java/io/quarkus/smallrye/restclient/deployment/SmallRyeRestClientProcessor.java

```diff
@@ -216,14 +224,29 @@ public void register(RegistrationContext registrationContext) {
 
     @BuildStep
     void registerProviders(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
-            JaxrsProvidersToRegisterBuildItem jaxrsProvidersToRegisterBuildItem) {
+            JaxrsProvidersToRegisterBuildItem jaxrsProvidersToRegisterBuildItem, CombinedIndexBuildItem combinedIndexBuildItem) {
         RestClientBuilderImpl.REGISTER_BUILTIN_PROVIDERS = jaxrsProvidersToRegisterBuildItem.useBuiltIn();
         RestClientBuilderImpl.PROVIDERS_TO_REGISTER = String.join(",", jaxrsProvidersToRegisterBuildItem.getProviders());
 
         // register the providers for reflection
         for (String providerToRegister : jaxrsProvidersToRegisterBuildItem.getProviders()) {
             reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, providerToRegister));
         }
+
+        // now we register all values of @RegisterProvider for constructor reflection
+
+        IndexView index = combinedIndexBuildItem.getIndex();
+        List<AnnotationInstance> allInstances = new ArrayList<>(index.getAnnotations(REGISTER_PROVIDER));
+        for (AnnotationInstance annotation : index.getAnnotations(REGISTER_PROVIDERS)) {
+            allInstances.addAll(Arrays.asList(annotation.value().asNestedArray()));
+        }
+        Set<String> providers = new HashSet<>();
```

Comment:
Any reason for the intermediary `providers`?

What value does it add over just iterating over `allInstances`?

Reply:
I wanted to avoid potentially registering the same provider multiple times. Do you think we shouldn't be worrying about that?

Reply:
@stuartwdouglas or @dmlloyd should be able to confirm, but I would imagine that because there could be duplicate classes added to reflection from different extensions, that the core de-dupes them for us.

Could be wrong though

Reply:
I'll obviously change it if the core doesn't dedup

Reply:
Looks like the core does dedup: https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateAutoFeatureStep.java#L317. I will proceed to remove the superfluous code

Reply:
Removed